<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsRasterCalcDialogBase</class>
 <widget class="QDialog" name="QgsRasterCalcDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>931</width>
    <height>576</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Raster Calculator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QgsScrollArea" name="scrollArea">
     <property name="frameShape">
      <enum>QFrame::Shape::NoFrame</enum>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>913</width>
        <height>502</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSplitter" name="splitter_2">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <widget class="QGroupBox" name="mRasterBandsGroupBox">
           <property name="title">
            <string>Raster Bands</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QListWidget" name="mRasterBandsListWidget"/>
            </item>
           </layout>
          </widget>
          <widget class="QGroupBox" name="mResultGroupBox">
           <property name="title">
            <string>Result Layer</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4" columnstretch="0">
            <item row="0" column="0">
             <widget class="QgsScrollArea" name="scrollArea_2">
              <property name="frameShape">
               <enum>QFrame::Shape::NoFrame</enum>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents_2">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>-153</y>
                 <width>493</width>
                 <height>334</height>
                </rect>
               </property>
               <layout class="QGridLayout" name="gridLayout_3">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item row="5" column="0" colspan="2">
                 <widget class="QGroupBox" name="groupBox_3">
                  <property name="title">
                   <string>Resolution</string>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout">
                   <item>
                    <widget class="QLabel" name="mColumnsLabel">
                     <property name="text">
                      <string>Columns</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QgsSpinBox" name="mNColumnsSpinBox">
                     <property name="maximum">
                      <number>999999999</number>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <spacer name="horizontalSpacer">
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>40</width>
                       <height>20</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                   <item>
                    <widget class="QLabel" name="mRowsLabel">
                     <property name="text">
                      <string>Rows</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QgsSpinBox" name="mNRowsSpinBox">
                     <property name="maximum">
                      <number>999999999</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="mOutputFormatLabel">
                  <property name="text">
                   <string>Output format</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QgsFileWidget" name="mOutputLayer" native="true"/>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="mVirtualLayerName">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="placeholderText">
                   <string>Autogenerated from expression</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0" colspan="2">
                 <widget class="QgsExtentGroupBox" name="mExtentGroupBox">
                  <property name="title">
                   <string>Spatial Extent</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QComboBox" name="mOutputFormatComboBox"/>
                </item>
                <item row="7" column="1">
                 <widget class="QgsProjectionSelectionWidget" name="mCrsSelector" native="true">
                  <property name="focusPolicy">
                   <enum>Qt::FocusPolicy::StrongFocus</enum>
                  </property>
                 </widget>
                </item>
                <item row="7" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Output CRS</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="mOutputLayerLabel">
                  <property name="text">
                   <string>Output layer</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="mVirtualLayerLabel">
                  <property name="text">
                   <string>Layer name</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0" colspan="2">
                 <widget class="QCheckBox" name="mUseVirtualProviderCheckBox">
                  <property name="toolTip">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Note: the result raster will not be usable with any processing algorithm that expects a materialized raster.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="layoutDirection">
                   <enum>Qt::LayoutDirection::LeftToRight</enum>
                  </property>
                  <property name="text">
                   <string>Create on-the-fly raster instead of writing layer to disk</string>
                  </property>
                  <property name="tristate">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
                <item row="9" column="0">
                 <widget class="QCheckBox" name="mAddResultToProjectCheckBox">
                  <property name="text">
                   <string>Add result to project</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item row="8" column="0" colspan="2">
                 <widget class="QgsCollapsibleGroupBox" name="mCreationOptionsGroupBox">
                  <property name="title">
                   <string>Create Options</string>
                  </property>
                  <property name="checkable">
                   <bool>true</bool>
                  </property>
                  <property name="checked">
                   <bool>false</bool>
                  </property>
                  <property name="collapsed" stdset="0">
                   <bool>true</bool>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <item>
                    <widget class="QgsRasterFormatSaveOptionsWidget" name="mCreationOptionsWidget" native="true"/>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QgsCollapsibleGroupBox" name="mOperatorsGroupBox">
             <property name="title">
              <string>Operators</string>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item row="0" column="0">
               <widget class="QPushButton" name="mPlusPushButton">
                <property name="text">
                 <string notr="true">+</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="mMultiplyPushButton">
                <property name="text">
                 <string notr="true">*</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="mOpenBracketPushButton">
                <property name="text">
                 <string notr="true">(</string>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QPushButton" name="mMinButton">
                <property name="text">
                 <string notr="true">min</string>
                </property>
               </widget>
              </item>
              <item row="0" column="4">
               <widget class="QPushButton" name="mConditionalStatButton">
                <property name="text">
                 <string notr="true">IF</string>
                </property>
               </widget>
              </item>
              <item row="0" column="5">
               <widget class="QPushButton" name="mCosButton">
                <property name="text">
                 <string notr="true">cos</string>
                </property>
               </widget>
              </item>
              <item row="0" column="6">
               <widget class="QPushButton" name="mACosButton">
                <property name="text">
                 <string notr="true">acos</string>
                </property>
               </widget>
              </item>
              <item row="0" column="7">
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>5</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="mMinusPushButton">
                <property name="text">
                 <string notr="true">-</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="mDividePushButton">
                <property name="text">
                 <string notr="true">/</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QPushButton" name="mCloseBracketPushButton">
                <property name="text">
                 <string notr="true">)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QPushButton" name="mMaxButton">
                <property name="text">
                 <string notr="true">max</string>
                </property>
               </widget>
              </item>
              <item row="1" column="4">
               <widget class="QPushButton" name="mAndButton">
                <property name="text">
                 <string notr="true">AND</string>
                </property>
               </widget>
              </item>
              <item row="1" column="5">
               <widget class="QPushButton" name="mSinButton">
                <property name="text">
                 <string notr="true">sin</string>
                </property>
               </widget>
              </item>
              <item row="1" column="6">
               <widget class="QPushButton" name="mASinButton">
                <property name="text">
                 <string notr="true">asin</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="mLessButton">
                <property name="text">
                 <string notr="true">&lt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="mGreaterButton">
                <property name="text">
                 <string notr="true">&gt;</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QPushButton" name="mEqualButton">
                <property name="text">
                 <string notr="true">=</string>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QPushButton" name="mAbsButton">
                <property name="text">
                 <string notr="true">abs</string>
                </property>
               </widget>
              </item>
              <item row="2" column="4">
               <widget class="QPushButton" name="mOrButton">
                <property name="text">
                 <string notr="true">OR</string>
                </property>
               </widget>
              </item>
              <item row="2" column="5">
               <widget class="QPushButton" name="mTanButton">
                <property name="text">
                 <string notr="true">tan</string>
                </property>
               </widget>
              </item>
              <item row="2" column="6">
               <widget class="QPushButton" name="mATanButton">
                <property name="text">
                 <string notr="true">atan</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QPushButton" name="mLesserEqualButton">
                <property name="text">
                 <string notr="true">&lt;=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QPushButton" name="mGreaterEqualButton">
                <property name="text">
                 <string notr="true">&gt;=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QPushButton" name="mNotEqualButton">
                <property name="text">
                 <string notr="true">!=</string>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QPushButton" name="mExpButton">
                <property name="text">
                 <string notr="true">^</string>
                </property>
               </widget>
              </item>
              <item row="3" column="4">
               <widget class="QPushButton" name="mSqrtButton">
                <property name="text">
                 <string notr="true">sqrt</string>
                </property>
               </widget>
              </item>
              <item row="3" column="5">
               <widget class="QPushButton" name="mLogButton">
                <property name="text">
                 <string notr="true">log10</string>
                </property>
               </widget>
              </item>
              <item row="3" column="6">
               <widget class="QPushButton" name="mLnButton">
                <property name="text">
                 <string notr="true">ln</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Raster Calculator Expression</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QTextEdit" name="mExpressionTextEdit"/>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="mExpressionValidLabel">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="mButtonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Help|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsExtentGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgsextentgroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>qgsscrollarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsProjectionSelectionWidget</class>
   <extends>QWidget</extends>
   <header>qgsprojectionselectionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsRasterFormatSaveOptionsWidget</class>
   <extends>QWidget</extends>
   <header>qgsrasterformatsaveoptionswidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>mButtonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsRasterCalcDialogBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mButtonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsRasterCalcDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
